src/render/mod.rs

```diff
@@ -7,68 +13,52 @@ pub mod mask;
-
-/// Turn a tree into an XObject. Returns the name (= the name in the `Resources` dictionary) of
-/// the XObject
-pub fn tree_to_x_object(
+/// Write a tree into a stream. Assumes that the stream belongs to transparency group and has the
+/// right bounding boxes.
+pub fn tree_to_stream(
     tree: &Tree,
     writer: &mut PdfWriter,
+    content: &mut Content,
     ctx: &mut Context,
-) -> Rc<String> {
-    let x_ref = ctx.alloc_ref();
-    ctx.deferrer.push();
-
-    let mut child_content = Content::new();
-    child_content.save_state();
-    child_content.transform(ctx.get_viewbox_transform().as_array());
-
-    // The root of a tree is always a group, so we can directly iterate over the children
-    for el in tree.root.children() {
-        el.render(writer, &mut child_content, ctx);
-    }
-
-    child_content.restore_state();
-
-    let child_content_stream = ctx.finish_content(child_content);
-
-    let mut x_object = writer.form_xobject(x_ref, &child_content_stream);
-    ctx.deferrer.pop(&mut x_object.resources());
-
-    if ctx.options.compress {
-        x_object.filter(Filter::FlateDecode);
-    }
-
-    x_object.bbox(Rect::new(0.0, 0.0, ctx.size.width() as f32, ctx.size.height() as f32));
-    // Apply the base transform
-    x_object.matrix(ctx.initial_transform.as_array());
-    x_object.finish();
+    initial_transform: Transform,
+) {
+    content.save_state();
+    let initial_transform = initial_transform.pre_concat(ctx.get_view_box_transform());
+    content.transform(initial_transform.to_pdf_transform());
 
-    ctx.deferrer.add_x_object(x_ref)
+    tree.root.render(writer, content, ctx, initial_transform);
+    content.restore_state();
 }
 
 trait Render {
-    fn render(&self, writer: &mut PdfWriter, content: &mut Content, ctx: &mut Context);
+    fn render(
+        &self,
+        writer: &mut PdfWriter,
+        content: &mut Content,
+        ctx: &mut Context,
+        accumulated_transform: Transform,
+    );
 }
 
 impl Render for Node {
-    fn render(&self, writer: &mut PdfWriter, content: &mut Content, ctx: &mut Context) {
+    fn render(
+        &self,
+        writer: &mut PdfWriter,
+        content: &mut Content,
+        ctx: &mut Context,
+        accumulated_transform: Transform,
+    ) {
         match *self.borrow() {
             NodeKind::Path(ref path) => {
-                path::render(path, &plain_bbox(self), writer, content, ctx)
+                path::render(self, path, writer, content, ctx, accumulated_transform)
             }
             NodeKind::Group(ref group) => {
-                group::render(self, group, writer, content, ctx)
+                group::render(self, group, writer, content, ctx, accumulated_transform)
             }
             #[cfg(feature = "image")]
             NodeKind::Image(ref image) => image::render(image, writer, content, ctx),
-            _ => {}
+            //Texts should be converted beforehand.
+            _ => unreachable!(),
```

Comment:
It would be more convenient for me if we continue to ignore text here rather than panic. This way it's possible to run svg2pdf on an unconverted usvg Tree and then handle the text separately. See discussion in https://github.com/typst/svg2pdf/issues/21

Reply:
Right, sorry about that, I changed it back!

Reply:
Thanks!!